### PR TITLE
Added dynamic command

### PR DESCRIPTION
**Pull request by SergeShkurko**
### Main idea:
Make it possible to intercept not existing commands

### Example where it might be needed:
Delegation command names for running scripts using CLI

### CODE REVIEW

**Comment by SergeShkurko** [comment]
@natebosch can you review?

**Comment by natebosch** [comment]
I don't have strong opinions on the overall approach - I can give a review of the code, but we should probably find a reviewer who can give feedback at a higher level.

cc @aadilmaan @mit-mit

File: lib/command_runner.dart [hunk 1]
```diff
@@ -104,6 +105,21 @@ class CommandRunner<T> {
...
+  /// is an a simple wrapper for [addCommand]
+  void addCommands(Iterable<Command<T>> commands) {
+    for (final command in commands) {
+      addCommand(command);
+    }
+  }
+
+  /// Adds [DynamicCommand] a top-level command in case
+  /// there are no suitable registered commands
+  void setDynamicCommand(DynamicCommand<T> command) {
```
Comment (natebosch) [inline lib/command_runner.dart]: How about `set dynamicComment(DynamicCommand<T> command)` with usage like `runner.dynamicCommand = command;`

**Comment by mit-mit** [comment]
Closing as assumed stale (sorry this didn't make it to merging)

**Pull request closed without merging**